### PR TITLE
crimson/osd: pass unknown args to ConfigProxy::parse_args()

### PR DESCRIPTION
**Pull request by tchaikov**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @liu-chunmei @athanatos 

this change is a part of #26697

File: src/crimson/osd/main.cc [hunk 1]
```diff
@@ -21,33 +21,66 @@ void usage(const char* prog) {
...
+  std::partition_copy(
+    argv, argv + argc,
+    std::back_inserter(ceph_args),
+    std::back_inserter(app_args),
+    [begin=unknown_args.begin(),
+     end=unknown_args.end(),
+     &consume_next_arg](char* arg) {
+      if (std::find(begin, end, arg) != end) {
+        return true;
+        // ceph_argparse_early_args() and
```
Comment (rzarzynski) [inline src/crimson/osd/main.cc]: Wouldn't be more readable to have this comment under the `else if (std::strcmp(arg, "-c") == 0) { ... }` block?

File: src/crimson/osd/main.cc [hunk 2]
```diff
@@ -21,33 +21,66 @@ void usage(const char* prog) {
   generic_server_usage();
 }
 
+auto partition_args(seastar::app_template& app, int argc, char* argv[])
```
Comment (rzarzynski) [inline src/crimson/osd/main.cc]: If it's possible to take `argv` as `const char* []`, I would vote for it.

Comment (tchaikov) [reply src/crimson/osd/main.cc]: it tries to align with the definition of `main()`. per n4659, § 6.6.1, the `main` function is either 

- a function of `()` or
- a function of `(int, pointer to pointer to char)`.

so, normally we have `main(int argc, char** argv)` or `main(int argc, char* argv[])`. but i cannot find a document claiming that we can (or cannot define) `argv` as a "pointer to const char*".

GCC does not complain if i define main like
```c++
main(int, const char* argv[])
```

but i am inclined to stick with the more popular signature of `(int argc, char* argv[])` before i find more supporting material.

File: src/crimson/osd/main.cc (hunk 1 above)
Comment (tchaikov) [reply src/crimson/osd/main.cc]: fixed and repushed.

File: src/crimson/osd/main.cc [hunk 3]
```diff
@@ -21,33 +21,66 @@ void usage(const char* prog) {
...
+                                                      bpo::include_positional);
+  std::vector<const char*> ceph_args, app_args;
+  bool consume_next_arg = false;
+  std::partition_copy(
+    argv, argv + argc,
+    std::back_inserter(ceph_args),
+    std::back_inserter(app_args),
+    [begin=unknown_args.begin(),
+     end=unknown_args.end(),
+     &consume_next_arg](const char* arg) {
```
Comment (tchaikov) [inline src/crimson/osd/main.cc]: @rzarzynski i changed this argument to `const`.

File: src/crimson/osd/main.cc (hunk 2 above)
Comment (rzarzynski) [reply src/crimson/osd/main.cc]: Sure, I'm entirely fine with that. Approving.

**Pull request merged**